syntax = "proto3";
package dfserver.pb;
option go_package = "github.com/huo-ju/dfserver/pkg/pb";

service Dfapi {
  rpc RunTask(Input) returns (stream Output) {}
}

message Output {
    string  InputId            = 1;
    int32   Version            = 2;
    string  ProducerName       = 3;
    string  ProducerSign       = 4;
    string  MimeType           = 5;
    bytes   Data               = 6;
    bytes   Args               = 7;
    string  Error              = 8;
}

message Input {
    string InputId             = 1;
    string Name                = 2;
    bytes  Settings            = 3;
    bytes  Data               = 4;
}

message Task {
    string  TaskId             = 1;
    repeated Output OutputList          = 2;
    repeated Input InputList            = 3;
}
